src/Http/Session/Session.php

```diff
@@ -4,121 +4,48 @@
+    ) {
     }
 
-    public function setId(string $id): void
+    public function put(string $key, mixed $value): void
     {
-        if (session_id($id) === false) {
-            throw new RuntimeException('Failed to set the session id.');
-        }
+        $this->getSessionManager()->put($this->id, $key, $value);
     }
 
-    public function getName(): string
-    {
-        return session_name();
-    }
-
-    public function setName(string $name): void
-    {
-        if (session_name($name) === false) {
-            throw new RuntimeException('Failed to set the session name.');
-        }
-    }
-
-    public function getSessionHandler(): SessionHandlerInterface
-    {
-        return $this->sessionHandler;
-    }
-
-    public function setSessionHandler(SessionHandlerInterface $sessionHandler): void
-    {
-        if (session_status() === PHP_SESSION_ACTIVE || headers_sent()) {
-            return;
-        }
-
-        session_set_save_handler($sessionHandler, false);
-    }
-
-    public function isStarted(): bool
-    {
-        return $this->isStarted;
-    }
-
-    public function start(): void
-    {
-        if ($this->isStarted()) {
-            return;
-        }
-
-        if (session_status() === PHP_SESSION_ACTIVE) {
-            throw new RuntimeException('The session has already been started by PHP.');
-        }
-
-        if (session_id() === false) {
-            $this->setId(session_create_id());
-        }
-
-        if (session_start() === false) {
-            throw new RuntimeException('Failed to start the session.');
-        }
-
-        $this->isStarted = true;
-    }
-
-    public function has(string $key): bool
-    {
-        return ArrayHelper::has($_SESSION, $key);
-    }
-
-    public function missing(string $key): bool
+    public function get(string $key, mixed $default = null): mixed
     {
-        return ! $this->has($key);
+        return $this->getSessionManager()->get($this->id, $key, $default);
     }
 
-    public function get(string $key, mixed $default = null): mixed
+    public function remove(string $key): void
     {
-        // We use an array helper here, so we can
-        // support dot notation.
-        return ArrayHelper::get($_SESSION, $key, $default);
+        $this->getSessionManager()->remove($this->id, $key);
     }
 
-    public function set(string $key, mixed $value): void
+    public function destroy(): void
     {
-        // We use an array helper here, so we can
-        // support dot notation.
-        ArrayHelper::set($_SESSION, $key, $value);
+        $this->getSessionManager()->destroy($this->id);
     }
 
-    public function all(): array
+    public function isValid(): bool
     {
-        return $_SESSION;
+        return $this->getSessionManager()->isValid($this->id);
     }
 
-    public function save(): void
+    private function getSessionManager(): SessionManager
     {
-        session_write_close();
+        return get(SessionManager::class);
```

Comment:
Why not inject this?

Reply:
Oh I know why, because we serialize the session itself.